### DISCUSSION TITLE
Checksum bug is a nightmare making Livewire very unstable

### DISCUSSION THREAD

**Discussion by andrei-todoran** [livewire/livewire#4120] [7. Bugs (archived)]
So I developed this project containing a couple of pages and everything was working fine for months (in local and staging/production environments)!

Then suddenly without changing any code got this "**Livewire encountered corrupt data when trying to hydrate the … component**" error.

I tried to apply different fixes, converting collections to arrays, try not to use numeric array keys, ordering array keys, removing 90% of the code and blade template content to make it easier to debug, etc. I even hardcoded very simple data with only one or two array elements(skipping reading from DB). Updated Livewire to latest version, cleared caches, configs, etc. 

Nothing helped, this error was keep coming back when clicking almost any button. Eventually I figured out the problem was caused by setting a public property to empty string(!). Something like:

`public $categoryName = "";`

then later:

`$this->categoryName = "";`

I had 2 or 3 variables like this which got set to empty string at some point. After setting them to NULL instead made the checksum error go away. I was at a point of giving up and converting the whole app to Vue. 

Hope this might help someone having the same issue. I wonder what makes Livewire not to like empty strings. I'm also wondering why this problem never appeared for months and suddenly just come up everywhere without any code change.

Livewire really needs to address these issues because it makes it unstable and unusable sometimes. Especially when you have to worry when the next "fingerprint" or "checksum" error is going to randomly pop up.

**Comment by joshhanley** [comment]
@andrei-todoran yeah the checksum issue is a tough one, but I think you'd agree ensuring Livewire is secure and users can't tamper with the request is important. Due to that though, there can be edge cases like this that unintentionally trigger the checksum failure.

If you are able to consistently reproduce the issue that is trigging the checksum, would you be able to create a small example repo which reproduces the issue with as little code as possible? Also can you confirm what version of Livewire you are using?

Please understand though, that this is an open source project, so saying things like "Livewire really needs to address these issues" isn't helpful. If something really needs fixing, then try and fix it and submit a PR. We welcome all PR's, but if that's not possible, at least a small clean working example of the bug is helpful or a failing test PR would be great, to make it easier for others to investigate.

Thanks!
